src/orchestrator.ts

```diff
@@ -550,28 +542,28 @@ export class Orchestrator {
         }
     }
 
-    private all(state: HistoryEvent[], tasks: Task[]): TaskSet {
-        const faulted = tasks.filter(TaskFilter.isFailedTask);
-        if (faulted.length > 0) {
-            return TaskFactory.FailedTaskSet(tasks, (faulted[0] as FailedSingleTask).exception);
-        }
-
-        const succesfulTasks = tasks.filter(TaskFilter.isSuccessfulSingleTask);
-        const isCompleted = succesfulTasks.length === tasks.length;
-        if (isCompleted) {
-            const results = succesfulTasks.reduce((acc, t) => {
-                return [...acc, t.result];
-            }, []);
+    private all(state: HistoryEvent[], tasks: TaskBase[]): TaskSet {
+        const completedTasks = tasks.filter(TaskFilter.isCompletedTask);
+        if (completedTasks.length === tasks.length) {
+            const sortedTasks = completedTasks.sort(TaskFilter.CompareFinishedTime);
+            const completionIndex = sortedTasks[tasks.length - 1].completionIndex;
 
-            return TaskFactory.SuccessfulTaskSet(tasks, results);
+            const failedTasks = completedTasks.filter(TaskFilter.isFailedTask);
```

Comment:
Should we use `sortedTasks` here instead of `completedTasks`, or does it not matter?

Reply:
It really shouldn't matter, at least after I finish the TODO below.